src/radical/pilot/pmgr/launching/default.py

```diff
@@ -241,60 +241,62 @@ def _handle_pilot_output_staging(self, pilot, sds):
+                if action in [rpc.COPY, rpc.LINK, rpc.MOVE]:
+                    raise ValueError("invalid pilot action '%s'" % action)
 
-            if action in [rpc.COPY, rpc.LINK, rpc.MOVE]:
-                self._prof.prof('staging_out_fail', uid=pid, msg=did)
-                raise ValueError("invalid action '%s' on pilot level" % action)
+                src = complete_url(src, src_context, self._log)
+                tgt = complete_url(tgt, tgt_context, self._log)
 
-            self._log.info('transfer %s to %s', src, tgt)
+                self._log.info('transfer %s to %s', src, tgt)
 
-            # FIXME: make sure that tgt URL points to the right resource
-            # FIXME: honor sd flags if given (recursive...)
-            flags = rsfs.CREATE_PARENTS
+                # FIXME: make sure that tgt URL points to the right resource
+                # FIXME: honor sd flags if given (recursive...)
+                flags = rsfs.CREATE_PARENTS
 
-            if os.path.isdir(src.path):
-                flags |= rsfs.RECURSIVE
+                if os.path.isdir(src.path):
+                    flags |= rsfs.RECURSIVE
 
-            # Define and open the staging directory for the pilot
+                # Define and open the staging directory for the pilot
 
-            # url used for cache (sandbox url w/o path)
-            tmp      = rs.Url(pilot['pilot_sandbox'])
-            tmp.path = '/'
-            key = str(tmp)
+                # url used for cache (sandbox url w/o path)
+                tmp      = rs.Url(pilot['pilot_sandbox'])
+                tmp.path = '/'
+                key = str(tmp)
 
-            self._log.debug("rs.file.Directory ('%s')", key)
+                with self._cache_lock:
+                    if key in self._saga_fs_cache:
+                        fs = self._saga_fs_cache[key]
 
-            with self._cache_lock:
-                if key in self._saga_fs_cache:
-                    fs = self._saga_fs_cache[key]
+                    else:
+                        fs = rsfs.Directory(key, session=self._session)
+                        self._saga_fs_cache[key] = fs
 
-                else:
-                    fs = rsfs.Directory(key, session=self._session)
-                    self._saga_fs_cache[key] = fs
+                fs.copy(src, tgt, flags=flags)
 
-            fs.copy(src, tgt, flags=flags)
+                sd['state'] = rps.DONE
+                self._prof.prof('staging_out_stop', uid=pid, msg=did)
 
-            sd['pmgr_state'] = rps.DONE
+            except:
+                self._prof.prof('staging_out_fail', uid=pid, msg=did)
+                sd['state'] = rps.FAILED
 
-            self.publish(rpc.CONTROL_PUBSUB, {'cmd': 'pilot_staging_output_result',
-                                              'arg': {'pilot': pilot,
-                                                      'sds'  : [sd]}})
 
-            self._prof.prof('staging_out_stop', uid=pid, msg=did)
```

Comment:
Why is this timestamp removed?

Reply:
It's now in line 298, and only recorded on success (as it should be)

Reply:
Missed it sorry.